### PR TITLE
fix(typescript): migrate ButtonGroup's stories and tests to typescript

### PR DESCRIPTION
**Pull request by ivanhdez28**

### CODE REVIEW

File: src/components/ButtonGroup/GroupButton/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import GroupButton from '.';
+import { noop } from '../../../util';
```
Comment (reywright) [inline src/components/ButtonGroup/GroupButton/index.test.tsx:5]: [should] this should be an absolute import.

File: src/components/ButtonGroup/GroupButton/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const Icon = () => <span>🌭</span>;
+const Wrapper = (props) => {
+  return <div {...props} data-test="snapshot-wrapper" />;
+};
```
Comment (reywright) [inline src/components/ButtonGroup/GroupButton/index.test.tsx:10]: [should] component should be removed

File: src/components/ButtonGroup/GroupButton/index.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ export interface GroupButtonProps {
    * `wrapper` must render the children prop passed into it
    * to render the GroupButton.
    */
-  Wrapper?: () => JSX.Element;
+  Wrapper?: (props: { children: React.ReactNode }) => JSX.Element;
```
Comment (reywright) [inline src/components/ButtonGroup/GroupButton/index.tsx:38]: [should] Wrapper and WrapperComponent should probably use a shared type.

File: src/components/ButtonGroup/index.test.tsx [hunk 4]
```diff
@@ -2,12 +2,13 @@ import React from 'react';
 import { render, screen, fireEvent } from '@testing-library/react';
 
 import ButtonGroup from '.';
+import { noop } from '../../util';
```
Comment (reywright) [inline src/components/ButtonGroup/index.test.tsx:5]: [should] absolute import

File: src/components/ButtonGroup/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import Tooltip from 'components/Tooltip';
+import FeedIcon from 'icons/react/FeedIcon';
+import State from 'util/storybook-docs/State';
+import { Story } from '@storybook/react';
+import ButtonGroup, { ButtonGroupProps } from '.';
+import { noop } from '../../util';
```
Comment (reywright) [inline src/components/ButtonGroup/index.stories.tsx:7]: [should] absolute import

**Review by reywright — CHANGES_REQUESTED** [review]
Thanks. I've made some requests on things I think should be changed, let me know what you think.

File: src/components/ButtonGroup/GroupButton/index.tsx (hunk 3 above)
Comment (ivanhdez28) [reply src/components/ButtonGroup/GroupButton/index.tsx:38]: can you elaborate on this? which one is WrapperComponent?

File: src/components/ButtonGroup/GroupButton/index.test.tsx (hunk 2 above)
Comment (reywright) [reply src/components/ButtonGroup/GroupButton/index.test.tsx:10]: Whoops 😞 I did not see that this was being used and being passed to `GroupButton`. That's why I was suggesting it's removal. My fault, disregard.

File: src/components/ButtonGroup/GroupButton/index.tsx (hunk 3 above)
Comment (reywright) [reply src/components/ButtonGroup/GroupButton/index.tsx:38]: Same file, up higher.

File: src/components/ButtonGroup/index.stories.tsx (hunk 5 above)
Comment (ivanhdez28) [reply src/components/ButtonGroup/index.stories.tsx]: fixed

File: src/components/ButtonGroup/index.test.tsx (hunk 4 above)
Comment (ivanhdez28) [reply src/components/ButtonGroup/index.test.tsx]: fixed

File: src/components/ButtonGroup/GroupButton/index.tsx (hunk 3 above)
Comment (ivanhdez28) [reply src/components/ButtonGroup/GroupButton/index.tsx]: fixed

File: src/components/ButtonGroup/GroupButton/index.test.tsx (hunk 1 above)
Comment (ivanhdez28) [reply src/components/ButtonGroup/GroupButton/index.test.tsx]: fixed.

**Review by reywright — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**